src/components/Send.tsx

```diff
@@ -1,4 +1,121 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import { withRouter, RouteComponentProps } from 'react-router';
+import { Form, FormGroup, Input, Label, Button } from 'reactstrap';
+import { getWallets } from '../store/wallet';
+import { AppState } from '../store';
+import api, { CreatePSBTOutputs } from '../api';
+import { Wallet } from '../types';
 
-// TODO: Send page
-export default (() => <>I'm the Send page</>) as React.SFC;
+interface DispatchProps {
+  getWallets: typeof getWallets;
+}
+
+interface ConnectProps {
+  // FIXME: typescript makes me accept possibility of nullness
+  // How can we just demand that this not be null for the component to even load?
+  activeWallet: Wallet | null;
+}
+
+type Props = DispatchProps & ConnectProps & RouteComponentProps;
+
+interface StateProps {
+  activeWallet: AppState['wallet']['activeWallet'];
+}
+
+interface LocalState {
+  outputs: CreatePSBTOutputs;
+  isSubmitting: boolean;
+  error: Error | null;
+}
+
+class Create extends React.Component<Props, LocalState> {
+  state: LocalState = {
+    outputs: [{'address': undefined, 'btc': undefined}],
+    isSubmitting: false,
+    error: null,
+  };
+  render() {
+    const { outputs } = this.state;
+
+    return (
+      <Form onSubmit={this.handleSubmit}>
+        {outputs.map((output, index) =>
+          <div>
+            <FormGroup>
+              <Label>Address</Label>
+              <Input
+                name="address"
+                value={output.address}
+                placeholder="..."
+                onChange={e => this.handleChange(e, index)}
+              />
+            </FormGroup>
+            <FormGroup>
+              <Label>Amount</Label>
+              <Input
+                name="btc"
+                value={output.btc}
+                type="number"
+                step="0.00000001"
+                placeholder="..."
+                onChange={e => this.handleChange(e, index)}
+              />
+            </FormGroup>
+          </div>
+        )}
+        <Button color="primary" size="lg" block onClick={this.handleAddOutput}>
+          Add Output
+        </Button>
+        <Button color="primary" size="lg" block>
+          Submit
+        </Button>
+      </Form>
+    );
+  }
+
+  private handleChange = (ev: React.ChangeEvent<HTMLInputElement>, index: number) => {
+    let { outputs } = this.state;
+    let target;
```

Comment:
Looking for a smarter way to implement this metho ...

Reply:
Typically I just keep input values as strings until I need to do logic or submit them. Otherwise, I'd probably just make it two different methods.